vivo-project/Vitro
# Role based search results filtering

# What does this pull request do?
New property will allow users to define filter values that could be used to exclude sensitive or technical information not intended for public display from  search results for any user roles defined in VIVO.

# What's new?
Replaced search:defaultPublic data property that was used to apply search filters to public role with
search:isDefaultForRole object property to specify role the filtering should be applied to.

# How should this be tested?
- Add [test_data.n3.zip](https://github.com/vivo-project/Vitro/files/13687215/test_data.n3.zip) to home/rdf/display/firsttime/
- Build VIVO
- If not logged in only continents should appear in search results
- If looged in as admin or self editor only continents should appear in search results
- If logged in as curator, editor or root, search results should not be limited to continents.

# Additional Notes:
Removed previously mistakenly committed file.
Default roles identifiers moved to VitroVocabulary 

# Interested parties
@VIVO-project/vivo-committers
